### PR TITLE
Boolean Toggle Tweaks

### PR DESCRIPTION
**Pull request by DrMaxis**
### Description

SDK updates for pvbid/app [#120](https://github.com/pvbid/app/pull/120)

### CODE REVIEW

File: src/domain/services/FieldAutoPopulateService.js [hunk 1]
```diff
@@ -132,12 +143,12 @@ export default class FieldAutoPopulateService {
     // if dep value is undefined
     if (_.isNull(dependencyValue) && _.isUndefined(this._field._data.value)) {
       this._field.config.is_auto_selected = true;
-      this._field._data.value = "-1";
+      this._field._data.value = typeof undefined;
```
Comment (DrMaxis) [inline src/domain/services/FieldAutoPopulateService.js:146]: Added typeof undefined here to set the value to undefined while maintaining the property on the field. Setting this to just undefined or null causes issues when reading/ setting (either removes the property entirely or sets it as null which does not save correctly.)

File: src/domain/services/FieldAutoPopulateService.js [hunk 2]
```diff
@@ -364,4 +370,23 @@ export default class FieldAutoPopulateService {
...
+    if(noCond(booleanValue)) {
+      this._field.config.default_value = false;
+    }
+    if(nullCond(booleanValue)) {
+      this._field.config.default_value = 'null';
+    }
+
+    this._field.emit("updated");
+    this._field.dirty();
+  }
```
Comment (DrMaxis) [inline src/domain/services/FieldAutoPopulateService.js:391]: Helper function for the bid fields config modal. Setting the value to 'null' matches the way it is saved by the admin toggle directive when a choice is not selected.

File: src/domain/services/BidEntityRelationsHelper.js [hunk 3]
```diff
@@ -485,7 +485,7 @@ export default class BidEntityRelationsHelper {
       return 'No';
     }
 
-    if (value === '-1' || value === -1 || typeof value === "undefined") {
+    if ( value === "null" ||  value === null || value.length === 0) {
```
Comment (rreynier) [inline src/domain/services/BidEntityRelationsHelper.js:488]: The "-1" stuff you are replacing, I assume thats an artifact of storing undefined as -1 in your previous PRs?

**Review by rreynier — APPROVED** [review]
:+1:

File: src/domain/services/BidEntityRelationsHelper.js (hunk 3 above)
Comment (DrMaxis) [reply src/domain/services/BidEntityRelationsHelper.js:488]: Yessir. If we use "-1" then it will affect the calculation/ prediction toggle.

**Pull request merged**